### ISSUE TITLE
Allow not specifying parameters names when using `placeholder` templater

### ISSUE THREAD

**Issue by FBorowiec** [sqlfluff/sqlfluff#4139]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### Description

_First let me thank the creators and maintainers of `SQLFluff`!
I learned a lot about SQL thanks to you :)_

I am now using the following configuration inside a `pyproject.toml` (thanks a lot for supporting that format btw <3):

```toml
[tool.sqlfluff.core]
templater = "placeholder"
dialect = "postgres"

[tool.sqlfluff.templater.placeholder]
param_style = "pyformat"
```

But as far as I understood I need to specify all the parameters inside the `sql` queries I have in my project.
So for example the query:

```sql
INSERT INTO my_table (
    var1,
    var2)
VALUES (
    %(var1)s,
    %(var2)s);
```

would require me to update the `pyproject.toml` with the following lines:

```toml
[tool.sqlfluff.templater.placeholder]
param_style = "pyformat"
var1 = 1  # <-- Added a random value
var2 = 2  # <-- Added a random value
```

My use case though is that I load each `sql` query in `python` in the following manner:

```python
def execute_query(**kwargs) -> None:
  # kwargs could be = {"var1": 1, "var2": 2}

  # Connecting to the database with psycopg2
  db_connection = psycopg2.connect(
      host="hostname",
      dbname="database",
      port="5432",
      user="postgres",
      password="",
  )
  cursor = db_connection.cursor()

  with open("path/to/query.sql", "r", encoding="utf-8") as query_file:
      query: str = query_file.read()
      with db_connection:
          cursor.execute(query, kwargs)
          db_connection.commit()
```

And I don't know the values of each parameter beforehand.
If I don't specify them `SQLFluff` will crash with the following error:

```bash
"sqlfluff.core.errors.SQLTemplaterError: Failure in placeholder templating: 'var1'.
Have you configured your variables?"
```

Is there a way not to have to specify each of the parameters inside the
`pyproject.toml` file?

Thanks a lot!

### Use case

I would like to not have to specify each of the parameters inside the `pyproject.toml` file.

### Dialect

`postgres`

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by barrywhart** [comment]
We could modify the placeholder templater to handle missing values the same way the Jinja templater does: https://docs.sqlfluff.com/en/stable/configuration.html#interaction-with-ignore-templating

There may be an existing issue for this.

**Comment by barrywhart** [comment]
I couldn't find the issue I was thinking about, so perhaps it was discussed but never created. There _is_ a similar issue here: #2108

Don't bother looking closely at that issue, because I'd rather the `placeholder` behavior mimic the current behavior of the `jinja` templater. That issue predates the `jinja` changes.

**Comment by FBorowiec** [comment]
@barrywhart Thanks for your support! I tried adding the `ignore_templated_areas` and `ignore` options (separately and then joined):

```toml
[tool.sqlfluff.core]
templater = "placeholder"
sql_file_exts = ".sql"
dialect = "postgres"
ignore_templated_areas = true <- added
ignore = "templating" <- added
```

but still got `SQLFluff` crashing whenever I open a `sql` file with `%(var)s` templates. Did I configure that right?

**Comment by barrywhart** [comment]
The setting is available, but only meaningful with the other templaters.

Sorry for the confusion!

**Comment by FBorowiec** [comment]
I could give it shot. Would [sqlfluff/core/templaters/placeholder.py](https://github.com/sqlfluff/sqlfluff/blob/60bff11c7a815c18731b84a79eaef3aada198294/src/sqlfluff/core/templaters/placeholder.py) be the right starting point?
Do you have some guidelines?

**Comment by barrywhart** [comment]
Yes, I think you'd be making changes in this area: https://github.com/sqlfluff/sqlfluff/blob/60bff11c7a815c18731b84a79eaef3aada198294/src/sqlfluff/core/templaters/placeholder.py#L154

These pages have some general info on contributions:
* https://github.com/sqlfluff/sqlfluff/blob/main/CONTRIBUTING.md
* https://github.com/sqlfluff/sqlfluff/wiki

**Comment by markoboy** [comment]
I had the same issue, although I'm not able to contribute into opening a PR at the moment. I'm not a python dev, so it will be hard for me to add any relevant tests and run all the docker stuff without exploring python further.

I've manually updated the script within the python lib within windows (mac/linux might be different path):
```python
# C:\Users\<YOUR_USER>\AppData\Local\Programs\Python\Python311\Lib\site-packages\sqlfluff\core\templaters\placeholder.py
# @ line 114
  if "ignore" in live_context:
    live_context["ignore_templating"] = loaded_context["ignore"] == "templating"
  else:
    live_context["ignore_templating"] = False

  return live_context

# @ line 155
  if context["ignore_templating"]:
    param_name = "__1"
    context["__1"] = "1"
  elif "param_name" not in found_param.groupdict():
```

Adding the following config then would allow to setup the placeholder without variables:
```properties
[sqlfluff:templater:placeholder]
param_regex = (\?{1,2})
ignore = templating
```

The `param_regex` is just for me into adding a `?` or `??` for handling knex bindings.

Thanks for this package, I'll try to introduce it to my team to lint our sql files as it seems to be quite powerful.
